### ISSUE TITLE
Casting PyArrow String Column to 'str' Returns 'object' Column w/ Wrong Values

### ISSUE THREAD

**Issue by kmcentush** [vaexio/vaex#1000]
**Description**
See title. In short, if you try to cast an existing string column to a 'str' datatype, the result is an object column that doesn't convert properly.

**Software information**
 - Vaex version (`import vaex; vaex.__version__)`:
```
{'vaex': '3.0.0',
 'vaex-core': '4.0.0a3',
 'vaex-viz': '0.4.0',
 'vaex-hdf5': '0.6.0',
 'vaex-server': '0.3.1',
 'vaex-astro': '0.7.0',
 'vaex-jupyter': '0.5.2',
 'vaex-ml': '0.9.0',
 'vaex-arrow': '0.5.1'}
```
 - Vaex was installed via: pip
 - OS: Windows 10

**Additional information**
Here's a code example:
```
import pyarrow as pa
import vaex

dt = vaex.from_arrays(a=['test', None])
print(dt['a'].dtype == pa.string())  # is a PyArrow string
print(dt)

dt['b'] = dt['a'].astype('str')
print(dt['b'].dtype == 'O')  # is an object
print(dt)

True
  #  a
  0  'test'
  1  --
True
  #  a       b
  0  'test'  None
  1  --      None
```

**Comment by JovanVeljanoski** [comment]
Hi, 

I am not 100% sure here (@maartenbreddels clarify better), but I think string types in vaex is meant to be arrow, and is not mean to be converted something else.

Is there any particular reason you are doing this conversion?

In any case, this behaviour is fixed in the new alpha ( i don't see a problem on master for example). 

Although I am surprised to see that a string type cast to string type is treated as an object on some level (@maartenbreddels )


<img width="601" alt="image" src="https://user-images.githubusercontent.com/18574951/95046439-7c245c80-06e4-11eb-9e60-b65ff8b9400a.png">

**Comment by kmcentush** [comment]
Hi @JovanVeljanoski,

I'm on vaex-core 4.0.0a3 and seeing this issue. What versions are your other vaex libraries? See my original post for more details on the specific versions I'm running.

In my specific use case, I'm writing an API. Given that some results may include None, etc, I'm forcing a type conversion so that downstream code works. My actual intention here is to convert the column to whatever string type is natively used by vaex. However, in attempting to do so by using "str", I ran into this bug.

I think one useful follow-up would be to include the supported dtypes for `astype()` in the docs. I had trouble figuring out what I should be passing in to convert to float (turns out it was `'float32' or 'float64`), string, etc. That's really the heart of my issue, but again, when I found this bug on my vaex versions I figured it'd be best to report it. :)

**Comment by JovanVeljanoski** [comment]
Hi, 

Ok, so when you put data into vaex.. vaex will take care of things. For example, once the data is recognized as string, you do not need  (and should not need) to case it again as string. Vaex uses arrow for anything to do with strings now. 

So in your OP, you are already converting a string type into numpy str i guess. 
You can always do `df.dtypes` to see what is the type of each column. 

If you want to get data out of vaex, for other standard python work, you can do things like (assuming you example above):
`df.a.tolist()` or `df.a.to_numpy()` 

Yeah, good point about the types `astype()` expects, it should be mentioned somewhere in the docs. 
In essence you can put any primitive type from numpy but as a string for example:
`.astype('int8')`
`.astype('float')`
`.astype('datetime64')`

In the future it will also support some if not all types from arrow (but @maartenbreddels  can explain this better).

If you are curious, this is my environment (from source, current master)
```
'vaex': '4.0.0.dev0',
 'vaex-core': '4.0.0a3',
 'vaex-viz': '0.5.0.dev0',
 'vaex-hdf5': '0.7.0a1',
 'vaex-server': '0.4.0.dev0',
 'vaex-astro': '0.8.0.dev0',
 'vaex-ui': '0.3.0',
 'vaex-jupyter': '0.6.0.dev0',
 'vaex-ml': '0.11.0a3',
 'vaex-graphql': '0.2.0.dev0'
```

I hope this helps!

**Comment by kmcentush** [comment]
Understood regarding Vaex casting appropriately when it loads the data.

To be clear, how would I cast an column of floats to string? You showed some examples of numpy primitive types, but based on their [docs](https://numpy.org/devdocs/user/basics.types.html#array-types-and-conversions-between-types) they don't explicitly note any string types. 

Specifically, how would I convert this column to a pyarrow string in Vaex? Yes, this example is impractical. But it illustrates my question well.
```
import vaex

dt = vaex.from_arrays(a=[1.0, -2.1, None])

# Expected result (where a is now of type 'pyarrow.string()')
#   a
0   '1.0'
1   '-2.1'
2   --
```

**Comment by JovanVeljanoski** [comment]
Yeah, so in  your example, you are passing data as dtype object. That, while supported in some way, is considered bad practice as far as vaex is concerned. Another thing I should mention is that (while I do it all the time), i think it is best not to pass Python lists in `vaex.from_arrays`. As the name says, it is expecting (numpy) arrays. Lists work, but they may result in objects without you knowing (as in the current case).

Best to pass a data of a single type.

If your problem is missing values, you have a couple of options. If working with floats, you can use np.nan, or a numpy masked arrays. `None` is not a float, so it creates a dtype object. Keep this in mind.

This is a good point, and we need to have a guide on this in the docs (@maartenbreddels I will take this on when I have a chance, maybe even this week).

As for your question, simply:
```
df = vaex.from_arrays(a=[1.0, -2.1, np.nan])  # so you get dtype float, None creates an object.
df['a_str'] = df.a.astype('string')
# df['a_str'] = df.a.astype('str')  should be the same i think
```

Note that in this case, the `np.nan` value will be converted to string so it will be `"nan"`, as if you did `str(np.nan)`

**Comment by kmcentush** [comment]
Thanks for the feedback. I'll leave this issue open as a placeholder for the added dtype section in the docs. Feel free to close this if you want to track that feature request elsewhere.

**Comment by kmcentush** [comment]
@JovanVeljanoski I also want to emphasize how cool it would be if Vaex kept masked (i.e. missing) values as missing during type conversions. Maybe this should be a parameter to `astype()` so the user can decide? Currently, vaex (or numpy/pyarrow) either errors on or fills the missing value. See some examples below.

```
# Generate the input
dt = vaex.from_arrays(
    int_test=np.ma.array(['1', '-1', ''], mask=[False, False, True], dtype=str),
    num_test=np.ma.array(['0.01', '', '-0.01'], mask=[False, True, False], dtype=str),
    string_test=np.ma.array(['test', 'test2', ''], mask=[False, False, True], dtype=str),
)

print(dt.dtypes)
print(dt)

# Output
int_test       string
num_test       string
string_test    string
dtype: object
  #  int_test    num_test    string_test
  0  1           0.01        test
  1  -1          --          test2
  2  --          -0.01       --
```

And the corresponding calls that throw errors:
```
# Errors
dt['int_test'] = dt['int_test'].astype('int64')
dt['num_test'] = dt['num_test'].astype('float64')
dt['string_test'] = dt['string_test'].astype('string')

print(dt.dtypes)
print(dt)

# Output
# <a long traceback w/ "ValueError: invalid literal for int() with base 10: <empty str>''
# <a long traceback also w/ "ValueError: could not convert string to float: <empty str>"
int_test              int64
num_test            float64
string_test    large_string
dtype: object
  #  int_test    num_test    string_test
  0  error       error       test
  1  error       error       test2
  2  error       error       None

# Desired output
int_test       int64
num_test       float64
string_test    string
dtype: object
#  int_test    num_test    string_test
  0  1           0.01        test
  1  -1          --          test2
  2  --          -0.01       --
```
